### PR TITLE
Pagination sorting swagger

### PR DESCRIPTION
**Pull request by Extase67**

### CODE REVIEW

File: src/main/java/com/example/demo/controller/BookController.java [hunk 1]
```diff
@@ -20,37 +24,44 @@
...
     @DeleteMapping
     @ResponseStatus(HttpStatus.NO_CONTENT)
+    @Operation(summary = "Delete book by id", description = "Delete book by id")
     public void deleteById(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update book by id", description = "Update book by id")
     public BookDto update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/controller/BookController.java]:
```suggestion
    public BookDto update(@PathVariable Long id, @RequestBody @Valid CreateBookRequestDto bookDto) {
```

File: src/main/java/com/example/demo/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+    @NotBlank
     private String title;
-
+    @NotBlank
     private String author;
-
+    @NotBlank
     private String isbn;
-
+    @Positive
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/dto/CreateBookRequestDto.java:19]:
```suggestion
    @Positive
    @NotNull
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: pom.xml [hunk 3]
```diff
@@ -72,6 +77,10 @@
             <artifactId>mapstruct</artifactId>
             <version>${mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
```
Comment (liliia-ponomarenko) [inline pom.xml]: better use spring-boot-starter-validation

File: src/main/java/com/example/demo/controller/BookController.java [hunk 4]
```diff
@@ -20,37 +24,44 @@
...
 @RequiredArgsConstructor
+@Tag(name = "Book management", description = "Endpoints for managing books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> findAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get list of all books")
+    public List<BookDto> findAll(Pageable pageable) {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/example/demo/controller/BookController.java]: better return Page instead of a List

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few recommendations ;)

**Pull request merged**